### ISSUE TITLE
CatBoostEncoder Transform method on validation data still showing missing values 

### ISSUE THREAD

**Issue by pratik-poudel** [scikit-learn-contrib/categorical-encoding#209]
After applying fit_transform() on train data and transform() on test data there is no missing values in  train data but still missing values in test data.

**Comment by janmotl** [comment]
Can you provide some example code? Something like:

```python
        enc = encoders.CatBoostEncoder(handle_missing='value')
        enc.fit(['a','a','b','c'], [1,1,1,0])
        test = enc.transform(['a','b',None])
        print(test)
```
?

**Comment by pratik-poudel** [comment]
fit_transform on a data frame having both categorical as well as numerical feature results is replacing all numeric columns missing values filled with -999 and doses not have any missing values on the train set.

`from category_encoders import  CatBoostEncoder`
`permutations = np.random.permutation(len(X_train))`
`predictors = X_train.iloc[permutations]`
`target = y_train.iloc[permutations]`
`cat_encoder = CatBoostEncoder(cols=categorical_columns)`
`X_train = cat_encoder.fit_transform(predictors, target)`

After transforming on the validation set, still there are missing values in the numeric columns.
`X_valid = cat_encoder.transform(X_validation)`

![cat_train](https://user-images.githubusercontent.com/37284545/64177639-a7080580-ce7f-11e9-8def-5b643e031151.png)

![cat_validation](https://user-images.githubusercontent.com/37284545/64177640-a7080580-ce7f-11e9-9a33-e6eb49f8e4f8.png)

**Comment by janmotl** [comment]
I see. It should be fixed now in the master branch.
